oecophylla/taxonomy/taxonomy.rule

```diff
@@ -123,6 +123,62 @@ rule taxonomy_kraken:
                   """)
 
 
+rule taxonomy_bracken:
+    """
+    Runs Kraken plus Bracken using general defaults.
+    """
+    input:
+        forward = qc_dir + "{sample}/filtered/{sample}.R1.trimmed.filtered.fastq.gz",
+        reverse = qc_dir + "{sample}/filtered/{sample}.R2.trimmed.filtered.fastq.gz"
+    output:
+        table = taxonomy_dir + "{sample}/bracken/{sample}_table.txt",
+        report = taxonomy_dir + "{sample}/bracken/{sample}_report.txt"
+    params:
+        kraken_env = config['envs']['kraken'],
+        kraken_db = config['params']['kraken']['kraken_db'],
+        bracken_db = config['params']['kraken']['bracken_db']
+    threads:
+        12
+    log:
+        taxonomy_dir + "logs/taxonomy_bracken.sample=[{sample}].log"
+    benchmark:
+        "benchmarks/taxonomy/taxonomy_bracken.sample=[{sample}].txt"
+    run:
+        with tempfile.TemporaryDirectory(dir=find_local_scratch(TMP_DIR_ROOT)) as temp_dir:
+            table_base = os.path.basename(output['table'])
+            report_base = os.path.basename(output['report'])
+
+            shell("""
+                  set +u; {params.kraken_env}; set -u
+
+                  kraken {input.forward} {input.reverse} \
+                    --db {params.kraken_db} \
+                    --paired \
+                    --fastq-input \
+                    --gzip-compressed \
+                    --only-classified-output \
+                    --threads {threads} \
+                    1> {temp_dir}/map.tmp \
+                    2> {log}
+
+                  kraken-report {temp_dir}/map.tmp \
+                    --db {params.kraken_db} \
+                    1> {temp_dir}/report.tmp \
+                    2>> {log}
+
+                  est_abundance.py -i {temp_dir}/report.tmp \
+                    -k {params.bracken_db} \
+                    -o {temp_dir}/{table_base} \
+                    2>> {log} 1>&2
+
+                  rm {temp_dir}/map.tmp
+                  rm {temp_dir}/report.tmp
+
+                  scp {temp_dir}/{table_base} {output.table}
```

Comment:
is `{temp_dir}/{table_base}` a directory, file, or pattern? if it is anything except a specific file, this copy will not work as expected

Reply:
This is a specific file, as defined by line #134.